lib/src/fnc/crypto.rs

```diff
@@ -182,11 +182,25 @@ pub mod scrypt {
 pub mod bcrypt {
 
 	use crate::err::Error;
+	use crate::fnc::crypto::COST_ALLOWANCE;
 	use crate::sql::value::Value;
 	use bcrypt;
+	use bcrypt::HashParts;
+	use std::str::FromStr;
 
 	pub fn cmp((hash, pass): (String, String)) -> Result<Value, Error> {
-		Ok(bcrypt::verify(pass, &hash).unwrap_or(false).into())
+		let parts = match HashParts::from_str(&hash) {
+			Ok(parts) => parts,
+			Err(_) => return Ok(Value::False),
+		};
+		Ok(if parts.get_cost() > bcrypt::DEFAULT_COST + COST_ALLOWANCE {
```

Comment:
Bcrypt cost is exponential, so add the cost allowance instead of multiplying.

Reply:
@finnbear could you add this in as a comment above line 196, just incase someone forgets about this down the line!

Reply:
Sure thing! I went ahead and added a similar comment to Scrypt, where the same concern applies to the `log_n` parameter.